pkg/mutation/mutate.go

```diff
@@ -2,44 +2,60 @@ package mutation
 
 import (
 	"bufio"
-	"encoding/json"
+	"bytes"
 	"fmt"
+	"strconv"
 	"strings"
 
-	jsonpatchV5 "github.com/evanphx/json-patch/v5"
 	"github.com/fairwindsops/polaris/pkg/config"
-	"github.com/fairwindsops/polaris/pkg/kube"
 	"github.com/fairwindsops/polaris/pkg/validator"
+	"github.com/pkg/errors"
 	"github.com/thoas/go-funk"
 	"gomodules.xyz/jsonpatch/v2"
+	yaml "gopkg.in/yaml.v3"
 )
 
-// ApplyAllSchemaMutations applies available mutation to a single resource
-func ApplyAllSchemaMutations(conf *config.Configuration, resourceProvider *kube.ResourceProvider, resource kube.GenericResource, mutations []jsonpatch.Operation) (kube.GenericResource, error) {
-	resByte := resource.OriginalObjectJSON
-	var jsonByte []byte
-	mutationByte, err := json.Marshal(mutations)
+// ApplyAllMutations applies available mutation to a single resource
+func ApplyAllMutations(manifest string, mutations []jsonpatch.Operation) (string, error) {
```

Comment:
It would be great to see some generic tests of this, in addition to the kubernetes ones. That way we can test really complex mutations, like indexing into multiple arrays.

E.g.
```yaml
pets:
- name: fido
  owners:
  - name: Alice
  - name: Bob
     aliases:
     - Robert
```

```
mutations:
- op: add
  path: /pets/0/owners/*/aliases
  value: Rob
```